dist/linux/appimage/build.sh

```diff
@@ -2,14 +2,16 @@
 
 cd $(dirname $0)
 REVISION_NO=`git rev-list --count HEAD`
+VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression=project.version -q -DforceStdout)
```

Comment:
1. Does this really work? Or do we need `-f ../../../pom.xml` so that the correct pom.xml is being used?
2. Instead of `org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate`, you could also just use `help:evaluate` so that no version number has to be specified.

Reply:
This vastly depends on the working dir. I believe we should discuss (thoroughly) where the scripts should be located relative to the projects root. And also document it in the README.md

Reply:
Oh, you're absolutely right. I made my own path error. And on a second look, it should also be _after_ the preconditions as well.
```suggestion
# check preconditions
if [ -z "${JAVA_HOME}" ]; then echo "JAVA_HOME not set. Run using JAVA_HOME=/path/to/jdk ./build.sh"; exit 1; fi
command -v mvn >/dev/null 2>&1 || { echo >&2 "mvn not found."; exit 1; }
command -v curl >/dev/null 2>&1 || { echo >&2 "curl not found."; exit 1; }

VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression=project.version -q -DforceStdout)
```

I agree with @overheadhunter as well though -- it really depends on where the scripts should be located. I don't have much to offer to that discussion, but as someone not well-versed in Java projects, the current structure made enough sense to me.

Reply:
> This vastly depends on the working dir.

Is it really? I thought `cd $(dirname $0)` would make sure that the working dir doesn't matter.

Reply:
Correct, but I'm not sure yet, if the current location is the best choice.